modules/github_reverse_proxy/main.tf

```diff
@@ -0,0 +1,163 @@
+}
+
+resource "aws_lambda_function_event_invoke_config" "concurrency_limit" {
+  function_name          = aws_lambda_alias.prod_alias.function_name
+  qualifier              = aws_lambda_alias.prod_alias.name
+  maximum_retry_attempts = 2
+}
+
+resource "aws_lambda_permission" "lambda_permission" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_alias.prod_alias.function_name
+  qualifier     = aws_lambda_alias.prod_alias.name
+  principal     = "apigateway.amazonaws.com"
+
+  # The /* part allows invocation from any stage, method and resource path
+  # within API Gateway.
+  source_arn = "${aws_api_gateway_rest_api.webhook_api.execution_arn}/*"
+}
+
+# API Gateway to act as a reverse proxy
+resource "aws_api_gateway_rest_api" "webhook_api" {
+  name = "GitHub Webhook Reverse Proxy"
+}
+
+# Create resource for webhooks
+resource "aws_api_gateway_resource" "webhook_resource" {
+  rest_api_id = aws_api_gateway_rest_api.webhook_api.id
+  parent_id   = aws_api_gateway_rest_api.webhook_api.root_resource_id
+  path_part   = "webhook"
+}
+
+# API Gateway Method
+resource "aws_api_gateway_method" "post_webhook" {
+  rest_api_id   = aws_api_gateway_rest_api.webhook_api.id
+  resource_id   = aws_api_gateway_resource.webhook_resource.id
+  http_method   = "POST"
+  authorization = "NONE"
+}
+
+# Lambda integration for API Gateway
+resource "aws_api_gateway_integration" "lambda_integration" {
+  rest_api_id             = aws_api_gateway_rest_api.webhook_api.id
+  resource_id             = aws_api_gateway_resource.webhook_resource.id
+  http_method             = aws_api_gateway_method.post_webhook.http_method
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = aws_lambda_alias.prod_alias.invoke_arn
+}
+
+# Deployment of API Gateway
+resource "aws_api_gateway_deployment" "api_deployment" {
+  rest_api_id = aws_api_gateway_rest_api.webhook_api.id
+  stage_name  = "prod"
+
+  depends_on = [aws_api_gateway_integration.lambda_integration]
+}
+
+resource "aws_api_gateway_rest_api_policy" "github_ip_restriction" {
+  rest_api_id = aws_api_gateway_rest_api.webhook_api.id
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+            {
+        Effect: "Allow",
+        Principal: "*",
+        Action: "execute-api:Invoke",
+        Resource: "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.webhook_api.id}/*/POST/*",
+      },
+      {
+        Effect: "Deny",
+        Principal: "*",
+        Action: "execute-api:Invoke",
+        Resource: "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.webhook_api.id}/*/POST/*",
+        Condition: {
+          "NotIpAddress": {
+            "aws:SourceIp": var.github_cidrs
```

Comment:
Exclude access from anything but github cidrs.

Since API Gateway/Lambda are serverless, it works slightly different. So no security group, as you maybe would expect.